src/algebra/big_operators.lean

```diff
@@ -773,6 +782,27 @@ begin
   exact add_lt_add_of_lt_of_le hlt (sum_le_sum $ λ j hj, Hle j  $ mem_of_mem_erase hj)
 end
 
+lemma sum_lt_sum_of_subset [decidable_eq α]
+  (h : s₁ ⊆ s₂) {i : α} (hi : i ∈ s₂ \ s₁) (hpos : 0 < f i) (hnonneg : ∀ j ∈ s₂ \ s₁, 0 ≤ f j) :
```

Comment:
Wouldn't it be better to ask for `(s₂ \ s₁).nonempty`?

Reply:
You want it to be nonempty, but you also want that `f` is strictly positive on an element of the difference, so in any case you will need to single out an element of the difference.